lib/Backend/LowerMDShared.cpp

```diff
@@ -8684,6 +8684,18 @@ void LowererMD::GenerateFastInlineBuiltInCall(IR::Instr* instr, IR::JnHelperMeth
         Assert(helperMethod == (IR::JnHelperMethod)0);
         return GenerateFastInlineBuiltInMathAbs(instr);
 
+    case Js::OpCode::InlineMathPow:
+#ifdef _M_IX86
+        if (!instr->GetSrc2()->IsFloat())
```

Comment:
If the exponent int32 type-spec can happen for both x86/x64, why restrict this to x86 ?